File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        str_n = str(n)
        visited = set()
        while str_n != "1":
            if str_n in visited:
                return False

            visited.add(str_n)
            str_n = str(sum(pow(int(digit), 2) for digit in str_n))
        
        return True

File: 202-happy-number/happy-number.cs
public class Solution {
    public bool IsHappy(int n) {
        HashSet<int> seen = new HashSet<int>();

        int num = n;
        while (!seen.Contains(num))
        {
            seen.Add(num);

            num = num.ToString().Sum(c => (int)Math.Pow(c - '0', 2));
            if (num == 1)
                return true;
        }

        return false;
    }
}